volatility3/framework/symbols/windows/extensions/consoles.py

```diff
@@ -0,0 +1,439 @@
+
+        seen = set()
+
+        for screen in screens:
+            cur = screen
+            while cur and cur.vol.offset != 0 and cur.vol.offset not in seen:
+                cur.TextBufferInfo.BufferRows.Rows.count = (
+                    cur.TextBufferInfo.BufferCapacity
+                )
+                yield cur
+                seen.add(cur.vol.offset)
+                cur = cur.Next
+
+    def get_histories(self):
+        for cmd_hist in self.HistoryList.dereference().to_list(
+            f"{self.get_symbol_table_name()}{constants.BANG}_COMMAND_HISTORY",
+            "ListEntry",
+        ):
+            yield cmd_hist
+
+    def get_exe_aliases(self):
+        exe_alias_list = self.ExeAliasList
+        # Windows 10 22000 and Server 20348 made this a Pointer
+        if isinstance(exe_alias_list, objects.Pointer):
+            exe_alias_list = exe_alias_list.dereference()
+        for exe_alias_list_item in exe_alias_list.to_list(
+            f"{self.get_symbol_table_name()}{constants.BANG}_EXE_ALIAS_LIST",
+            "ListEntry",
+        ):
+            yield exe_alias_list_item
+
+    def get_processes(self):
+        for proc in self.ConsoleProcessList.dereference().to_list(
+            f"{self.get_symbol_table_name()}{constants.BANG}_CONSOLE_PROCESS_LIST",
+            "ListEntry",
+        ):
+            yield proc
+
+    def get_title(self):
+        try:
+            return self.Title.dereference().cast(
+                "string", encoding="utf-16", errors="replace", max_length=512
+            )
+        except:
+            return ""
+
+    def get_original_title(self):
+        try:
+            return self.OriginalTitle.dereference().cast(
+                "string", encoding="utf-16", errors="replace", max_length=512
+            )
+        except:
+            return ""
+
+
+class COMMAND(objects.StructType):
+    """A Command Structure"""
+
+    def is_valid(self):
+        if (
+            self.Length < 1
+            or self.Allocated < 1
+            or self.Length > 1024
+            or self.Allocated > 1024
+        ):
+            return False
+
+        return True
+
+    def get_command(self):
+        if self.Length < 8:
+            return self.Chars.cast(
+                "string",
+                encoding="utf-16",
+                errors="replace",
+                max_length=self.Length * 2,
+            )
+        elif self.Length < 1024:
+            return self.Pointer.dereference().cast(
+                "string", encoding="utf-16", errors="replace", max_length=512
+            )
+
+
+class COMMAND_HISTORY(objects.StructType):
+    """A Command History Structure."""
+
+    @property
+    def CommandCount(self):
+        command_type = self.get_symbol_table_name() + constants.BANG + "_COMMAND"
+        command_size = self._context.symbol_space.get_type(command_type).size
+        return int((self.CommandBucket.End - self.CommandBucket.Begin) / command_size)
```

Comment:
If `command_size` ever isn't a multiple of the total size, this could end up rounding down.  If we're sure this'll always divide properly to provide an integer value, that's fine but if not it'll need tweaking.

Reply:
it should be a multiple.